transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoopGroup.java

```diff
@@ -146,9 +155,25 @@ public void setIoRatio(int ioRatio) {
 
     @Override
     protected EventLoop newChild(Executor executor, Object... args) throws Exception {
-        EventLoopTaskQueueFactory queueFactory = args.length == 4 ? (EventLoopTaskQueueFactory) args[3] : null;
-        return new EpollEventLoop(this, executor, (Integer) args[0],
-                ((SelectStrategyFactory) args[1]).newSelectStrategy(),
-                (RejectedExecutionHandler) args[2], queueFactory);
+        Integer maxEvents = (Integer) args[0];
+        SelectStrategyFactory selectStrategyFactory = (SelectStrategyFactory) args[1];
+        RejectedExecutionHandler rejectedExecutionHandler = (RejectedExecutionHandler) args[2];
+        EventLoopTaskQueueFactory taskQueueFactory = null;
+        EventLoopTaskQueueFactory tailTaskQueueFactory = null;
+
+        switch (args.length) {
+            case 4:
```

Comment:
nit: using `if (args.length > 3)` will remove redundancy for common arguments:

```java
if (args.length > 3) {
    taskQueueFactory = (EventLoopTaskQueueFactory) args[3];
}
if (args.length > 4) {
    tailTaskQueueFactory = (EventLoopTaskQueueFactory) args[4];
}
```

Reply:
There is always from 3 to 5 arguments for `EventLoopGroups` that exist in this project, so my thinking was that current form is simpler. Conditionals form looked better where there is wider range of optional arguments (>2) - can change if It is more convenient.

Reply:
Sounds good, just a nit so feel free to ignore.